src/robot/libdocpkg/jsonwriter.py

```diff
@@ -0,0 +1,30 @@
+#  Copyright 2008-2015 Nokia Networks
+#  Copyright 2016-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+
+from robot.utils import file_writer
+
+
+class LibdocJsonWriter(object):
+
+    def __init__(self, spec_doc_format):
+        self._spec_doc_format = spec_doc_format
+
+    def write(self, libdoc, outfile):
+        if self._spec_doc_format == 'HTML':
+            libdoc.convert_doc_to_html()
```

Comment:
I think it would be a bit cleaner if the caller would have called `libdoc.convert_doc_to_html()` if needed already. Then there would be no need to pass `spec_doc_format` here or to other writers.

Reply:
you can refactor if you really want, but i think that is easy to understand.

Reply:
that is not invented in that task, so i would like to keep jsonwriter as all other writers